src/utilities/activities.ts

```diff
@@ -163,3 +167,82 @@ export function preprocessActivityDirectiveDB(
   }
   return { ...activityDirectiveDB, start_time_ms };
 }
+
+export function copyActivityDirectivesToClipboard(sourcePlan: Plan, activities: ActivityDirective[]) {
+  const copiedActivityIds = new Set(activities.map(a => a.id));
+  const clippedActivities = activities.map(activity => {
+    const anchorInSelection = activity.anchor_id !== null && copiedActivityIds.has(activity.anchor_id);
+    return {
+      anchor_id: anchorInSelection ? activity.anchor_id : null,
+      anchored_to_start: activity.anchored_to_start,
+      arguments: activity.arguments,
+      id: activity.id,
+      name: activity.name,
+      start_offset: activity.start_offset,
+      start_time_ms: activity.start_time_ms,
+      tags: activity.tags,
+      type: activity.type,
+    };
+  });
+
+  const clipboard = {
+    activities: clippedActivities,
+    sourcePlanId: sourcePlan.id,
+  };
+
+  setClipboardContent(JSON.stringify(clipboard));
+  showSuccessToast(`Copied ${activities.length} Activity Directive${activities.length ? '' : 's'}`);
+}
+
+export function canPasteActivityDirectivesFromClipboard(destinationPlan: Plan | null): boolean {
+  if (destinationPlan === null) {
+    return false;
+  }
+
+  const serializedClipboard = getClipboardContent();
+  if (serializedClipboard === null) {
+    return false;
+  }
+
+  const clipboard = JSON.parse(serializedClipboard);
```

Comment:
Consider try/catching instances of parsing from session storage in case the JSON blob gets corrupted _somehow_ (know it is unlikely but might be possible if a user refreshes the page while writing a huge set of activities to session storage).

Reply:
Good checks